oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/tool/AzureCheck.java

```
@@ -388,7 +389,13 @@ public int run() {
             persistence = ToolUtils.decorateWithCache(persistence, persistentCachePath, persistentCacheSizeGb);
         }
 
-        FileStoreBuilder builder = fileStoreBuilder(Files.createTempDir()).withCustomPersistence(persistence);
+        FileStoreBuilder builder = null;
+        try {
+            builder = fileStoreBuilder(Files.createTempDirectory(getClass().getSimpleName() + "-").toFile()).withCustomPersistence(persistence);
+        } catch (IOException e) {
+            e.printStackTrace(err);
+            return 1;
+        }
```

Maybe throw an IllegalStateException instead of returning 1 to keep closest to the Guava behavior?

---

This is part of a commend line tool.

Look at the end of the method; this is exactly what it's already doing; the only new thing is the creation of the temp directory can fail.

---

I don't understand, the Guava version may also fail, just with a RuntimeException.